rflx/identifier.py

```diff
@@ -20,10 +20,16 @@ def __init__(
         else:
             assert False, f'unexpected identifier type "{type(identifier).__name__}"'
 
-        assert self._parts, "empty identifier"
-        assert "" not in self._parts, "empty part in identifier"
-        for c in [" ", ".", ":"]:
-            assert all(c not in part for part in self._parts), f'"{c}" in identifier parts'
+        error = RecordFluxError()
+        if not self._parts:
+            error.append("empty identifier", Subsystem.ID, Severity.ERROR)
+        elif "" in self._parts:
+            error.append("empty part in identifier", Subsystem.ID, Severity.ERROR)
+        else:
+            for c in [" ", ".", ":"]:
+                if not all(c not in part for part in self._parts):
```

Comment:
`if any(c in part ...` would be simpler than `if not all(c not in part ...`.

Reply:
Changed in 071a224